sftpcloudfs/main.py

```diff
@@ -416,10 +415,16 @@ def run(self):
         dc.pidfile = self.pidfile
 
         if self.options.uid:
-            dc.uid = self.options.uid
+            if isinstance(self.options.uid, basestring):
+                dc.uid = pwd.getpwnam(self.options.uid).pw_uid
```

Comment:
This can raise a `KeyError`. We should capture that and provide an user friendly error.

Reply:
Done. I had to move the `self.setup_log()` call to get `self.log` earlier.

Reply:
Not sure that's a good idea. The log must be setup after the daemon context has been set (IIRC).

Unless it works differently, I think we can't do this.

I would move your change to the method that is processing the options and uses `parser.error` to report issues. The place where you added the `pwd` support is to set the daemon context and it is a bit too late to process options.

Sorry, I should have seen this earlier.

Reply:
You're right, that a better solution !
I just pushed a updated commit implementing this method.